### PR TITLE
Adding a length property to GridGeoSampler

### PR DESCRIPTION
**Pull request by calebrob6**
Two changes in sampler land: 
- Adds a `__len__` to GridGeoSampler
- Factors out the logic for sampling a random bounding box from the extent of another bounding box

### CODE REVIEW

File: torchgeo/samplers/batch.py [hunk 1]
```diff
@@ -95,16 +95,9 @@ def __iter__(self) -> Iterator[List[BoundingBox]]:
             # Choose random indices within that tile
```
Comment (adamjstewart) [inline torchgeo/samplers/batch.py:95]: In the `__init__` method, I think we can skip calling `_to_tuple` now that we call it in the `get_random_bounding_box` method instead.

File: torchgeo/samplers/single.py [hunk 2]
```diff
@@ -185,3 +186,11 @@ def __iter__(self) -> Iterator[BoundingBox]:
                     maxx = minx + self.size[1]
 
                     yield BoundingBox(minx, maxx, miny, maxy, mint, maxt)
+
+    def __len__(self) -> int:
```
Comment (adamjstewart) [inline torchgeo/samplers/single.py:190]: What's the use case for having a `__len__` method?

File: torchgeo/samplers/batch.py (hunk 1 above)
Comment (calebrob6) [reply torchgeo/samplers/batch.py:95]: Sure

File: torchgeo/samplers/single.py (hunk 2 above)
Comment (calebrob6) [reply torchgeo/samplers/single.py:190]: I don't think I'm understanding your question correctly. The obvious use case is when a user wants to know how many patches will be sampled in the grid sampler... You can use this information to show progress bars, benchmark time per patch without having to keep a running count of the patches, to satisfy your curiosity, etc.

Comment (adamjstewart) [reply torchgeo/samplers/single.py:190]: Gotcha, I've never tried to add a progress bar before but it's not a bad idea. I think the reason I didn't add a `__len__` method in the first place was that I was worried about efficiency, but I think it's fine.

**Review by adamjstewart — DISMISSED** [review]

**Comment by calebrob6** [comment]
Removing the to tuple in the constructor breaks a bunch of tests, going to leave it in so we can assume that size is always a tuple

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**